content/blog/cozycoder-vision/index.md

```diff
@@ -0,0 +1,69 @@
+
+## Giving Developers more Control
+
+The obvious problem with chat is that everyone is just treating it as chat between a human and a bot. But that's not true. Each time you interact with Copilot or other LLMs the full conversation is send to the LLM as a context. There is no reason, why this context couldn't be treated like a Git history instead.
+
+Let's imagine a tool that gives developers more control and go through the above example:
+You start a conversation with a LLM to generate you new project. Of course, it will also run the generated code on your local machine automatically to prove it is working. Once your initial project setup is done, a pull request is provided that can be reviewed by your colleagues. Of course, your colleagues will have access to chat, which is linked in the pull request. Next time one of your colleagues wants to setup another project, they just fork your conversation at some point and continue a new conversation from that point on instead of talking to the LLM from scratch. Since new projects are created frequently in your company, someone is taking your conversation and introduces some parameters to it to use it as a template. Now you have a reusable conversation template that you can simply trigger from your terminal, provide few details (e.g. a project name) and get the code generated.
+
+Of course, not every company creates so many new projects. But this is just one example to show what could be possible, if we were using LLMs more efficiently.
+
+## Designed for your Laptop not the Cloud
+
+The above example gives a rough idea of what you as a software developer should be able to do with Cozy Coder. But to make the tool really good, it obviously must run on your laptop.
+
+Many developers use CLIs and they can easily be used within scripts to automate tasks. As Cozy Coder is designed for developers, it must provide a proper CLI. But also the access to the file system is important to directly modify local files, if you want to.
+
+Even more important are the constraints of large enterprises. In the last decade, I haven't worked for a single large company here in Europe that did not host their own Github Enterprise or Gitlab or similar tool. The code is capitalized and must to be kept secret. Hosting the code on a self-hosted Github Enterprise that is only accessible from your VPN is how you achieve this. So why should one of those companies start sending code to some LLM hosted in the cloud? Everything that runs locally on your laptop is significantly easier to use within larger enterprises.
+
+Cozy Coder will of course not be limited to local LLMs, but they will always be the preferred choice and well supported. Ideally we will all together achieve the usage of local LLMs, where possible and only use the cloud based ones, where necessary.
+
+Of course, Cozy Coder will also have a server side part. But it will be mainly for enabling sharing and collaboration as well backups of data.
```

Comment:
Double use of "of course". If I may suggest swapping one of them out for a synonym :)